src/rest/resources.ts

```diff
@@ -294,61 +307,59 @@ export class Resources implements IResources {
             method: 'GET',
             baseURL: this.baseUrl,
             url: `/oauth/token?oauth_provider=${oAuthProvider}`,
-            headers: this.getHeadersWithAuthorization(this.userToken)
+            headers: this.headers,
         });
     }
 
     public getOAuthProviders(): AxiosPromise<any[]> {
         return this.axios.request<any[]>({
             method: 'GET',
-            headers: this.getHeadersWithAuthorization(this.userToken),
+            headers: this.headers,
             baseURL: this.baseUrl,
-            url: '/oauth'
+            url: '/oauth',
         });
     }
 
-    public updateActivity(workspaceId: string): AxiosPromise<void> {
+    public updateActivity(workspaceId: string): AxiosPromise<void> { // -
         return this.axios.request<void>({
             method: 'PUT',
             baseURL: this.baseUrl,
             url: `/activity/${workspaceId}`,
-            headers: this.getHeadersWithAuthorization(this.machineToken)
+            headers: this.headers,
         });
     }
 
-    private getHeadersWithAuthorization(token?: string): { [key: string]: string } {
+    public getDevfileSchema<T>(): AxiosPromise<T> {
+        return this.axios.request<T>({
+            method: 'GET',
+            baseURL: this.baseUrl,
+            url: this.devfileUrl,
+            headers: this.headers,
+        });
+    }
+
+    public getKubernetesNamespace<T>(): AxiosPromise<T> {
+        return this.axios.request<T>({
+            method: 'GET',
+            baseURL: this.baseUrl,
+            url: '/kubernetes/namespace',
+            headers: this.headers,
+        });
+    }
+
+    private get headers(): { [key: string]: string } {
```

Comment:
Why do you prefer to recalculate headers each time this getter's called? Have you considered to calculate it once while class instantiating, or memoize the result?

Reply:
Just like an old implementation that generated each ti. I will change it.

Reply:
done